src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,79 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React, { useState } from 'react';
+import { getActiveTodos, getCompletedTodos, getTodos } from '../../api';
+import { Todo } from '../../types/Todo';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  changeQuery: (value: string) => void,
+  handleChangeTodos: (value: Promise<Todo[]>) => void,
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = ({
+  changeQuery,
+  handleChangeTodos,
+}) => {
+  const [searchValue, setSearchValue] = useState('');
+  const changeSearchValue = (value: string) => {
+    changeQuery(value);
+    setSearchValue(value);
+  };
+
+  enum TodoStatus {
+    All = 'all',
+    Active = 'active',
+    Completed = 'completed',
+  }
```

Comment:
Move the enum from the component
```suggestion
  enum TodoStatus {
    All = 'all',
    Active = 'active',
    Completed = 'completed',
  }
  
  export const TodoFilter: React.FC<Props> = ....
```

or even you can create a separate file for types